src/server/db/actions/UserAction.ts

```diff
@@ -56,3 +57,77 @@ export async function verifyUser(email: string, password: string) {
     _id: nonSensitiveUser._id.toString(),
   };
 }
+
+/**
+ * Gets a user by their ID.
+ * @param {string} id ID of the user to get.
+ * @throws {UserDoesNotExistException} If unable to find user
+ */
+export async function getUser(id: string) {
+  await connectMongoDB();
+  const user = await UserModel.findById(id);
+  if (!user) {
+    throw new UserDoesNotExistException();
+  }
+  return user;
+}
+
+/**
+ * Edits a user.
+ * @param {z.infer<typeof userSchema> & { _id: string }} userInfo Info of the user to find/update.
+ * @throws {GenericUserErrorException} If changing email to an email that corresponds to a pre-existing account
+ * @throws {UserDoesNotExistException} If unable to find user
+ * 
+ */
+export async function editUser(userInfo: z.infer<typeof userSchema> & { _id: string }) {
+  await connectMongoDB();
+
+  const existingUser = await UserModel.findOne({ email: userInfo.email });
+  
+  if (existingUser && String(existingUser._id) != userInfo._id) {
+    throw new UserAlreadyExistsException();
+  }
```

Comment:
Can you help me understand the second condition for this `if` statement? You can also do `existingUser.toObject()` instead of having to manually turn `existingUser`'s `_id` into a string

Reply:
it's basically a check for, if the inputted email already exists with a user(first condition), is that existing user the same as this user (2nd condition).  if not (id's are unequal), the user's trying to change their email to an email that already exists on a separate user, so the error is thrown. basically preventing multiple accounts from having the same email.

is there a better way of doing this? also, for using ```existingUser.toObject()```, do you mean to do something like ```existingUser.toObject()._id.toString()```?  online i've only been able to find people using ```._id.toString()```  or ```._id.valueOf()``` alongside ```existingUser.toObject()``` to get the id as a string.

Reply:
Oh, gotcha! I didn't even consider the case where they would be changing to an existing email. Nice catch!

Yep, you're totally right about `_id.toString()`. You'll need to do that alongside `toObject()`! I should've double checked before telling you that my bad